### PR TITLE
[SQL Server DevOps auditing policy] Support SQL server DevOps audit policy commands

### PR DESCRIPTION
**Pull request by Oleza1972**
In this PR the following commands have been introduced to support SQL server DevOps policy :

- az sql server ms-support audit-policy show

- az sql server ms-support audit-policy update

Examples of commands:

sql server ms-support audit-policy update -g my_resource_group -s my_server -n my_database --state Enabled --blob-storage-target-state Enabled --storage-endpoint https://mystorageaccount.blob.core.windows.net/

sql server ms-support  audit-policy update -g my_resource_group -s my_server -n my_database --state Enabled --blob-storage-target-state Disabled

sql server ms-support  audit-policy update -g my_resource_group -n my_server --state Enabled --log-analytics-target-state Enabled --log-analytics-workspace-resource-id /subscriptions/1111-2222-3333-444-5555555/resourcegroups/my_resource_group/providers/microsoft.operationalinsights/workspaces/my_workspace

sql server ms-support audit-policy update -g my_resource_group -n my_server --state Enabled --log-analytics-target-state Disabled --event-hub-target-state Enabled --event-hub-authorization-rule-id /subscriptions/1111-2222-3333-4444/resourceGroups/my_resource_group/providers/Microsoft.EventHub/namespaces/my_eventhub_namespace/authorizationrules/my_eventhub_auth_rule_name --event-hub my_event_hub_name

Testing Guide

Unit test test_sql_server_ms_support_mgmt covers the test for the change.

History Notes

[SQL] az sql server ms-support audit-policy show: show server's DevOps audit policy
[SQL] az sql server ms-support audit-policy update: update server's DevOps audit policy
[SQL] az sql server ms-support audit-policy wait: place the CLI in a waiting state until a condition of the server's DevOps audit policy is met.

### CODE REVIEW

**Comment by yonzhan** [comment]
sql

File: src/azure-cli/azure/cli/command_modules/sql/commands.py [hunk 1]
```diff
@@ -732,24 +749,6 @@ def load_command_table(self, _):
             is_preview=True)
         g.wait_command('wait')
 
-    with self.command_group('sql midb log-replay',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/commands.py:735]: May I know the reason why removing these commands?
If you don't want to support them any more, please follow deprecation policy.

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 2]
```diff
@@ -576,21 +576,21 @@ def test_sql_db_read_replica_mgmt(self, resource_group, resource_group_location,
                      JMESPathCheck('edition', edition),
                      JMESPathCheck('sku.tier', edition),
                      JMESPathCheck('readScale', 'Enabled'),
-                     JMESPathCheck('readReplicaCount', '1')])
+                     JMESPathCheck('readReplicaCount', 'None')])
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:579]: Is there any breaking change in server side?
Typically although we upgrade the api version, we need to make sure there is no breaking change in response.

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 3]
```diff
@@ -576,21 +576,21 @@ def test_sql_db_read_replica_mgmt(self, resource_group, resource_group_location,
...
-                     JMESPathCheck('readReplicaCount', '3')])
+                     JMESPathCheck('readReplicaCount', 'None')])
 
         # Decrease read replicas
         self.cmd('sql db update -g {} --server {} --name {} --read-replicas {}'
                  .format(resource_group, server, database_name, 0),
                  checks=[
-                     JMESPathCheck('readScale', 'Disabled'),
-                     JMESPathCheck('readReplicaCount', '0')])
+                     JMESPathCheck('readScale', 'Enabled'),
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:592]: same

**Comment by Juliehzl** [comment]
please use `azdev linter sql` to fix linter errors, `azdev style sql` to fix style errors and then use `azdev test sql` to make sure all th test pass.

If there is api version upgrade, you need rerecord the file.

**Comment by Juliehzl** [comment]
after discussion with other team, they say the new api version is not ready to release now.

**Comment by Oleza1972** [comment]
I created another branch server_devops_audit_support_2 which contains only DevOps changes.
It will wait for SDK release.

**Pull request closed without merging**